metadata_backend/api/handlers/object.py

```diff
@@ -253,19 +253,14 @@ async def delete_object(self, req: Request) -> web.HTTPNoContent:
 
         submission_op = SubmissionOperator(db_client)
         exists, submission_id, published = await submission_op.check_object_in_submission(collection, accession_id)
-        if exists:
-            if published:
-                reason = "published objects cannot be deleted."
-                LOG.error(reason)
-                raise web.HTTPUnauthorized(reason=reason)
-            await submission_op.remove_object(submission_id, collection, accession_id)
-            _now = int(datetime.now().timestamp())
-            lastModified = {"op": "replace", "path": "/lastModified", "value": _now}
-            await submission_op.update_submission(submission_id, [lastModified])
-        else:
-            reason = "This object does not seem to belong to any user."
+        if exists and published:
+            reason = "Published objects cannot be deleted."
             LOG.error(reason)
-            raise web.HTTPUnprocessableEntity(reason=reason)
```

Comment:
I didn't see a reason why the `else` portion could ever be reached when there is a `check_exists()` method call already right before this. There aren't even any tests that would reach this edge case currently. Hence the deletion. I could be misunderstanding though...

Reply:
Sounds like leftovers from a previous change.

Reply:
the reason was objects that were not attached to any submission - It was done as a sanity check in case the db gets altered and we have accessionIDs not attached to a submission or that are attached to more than 1 submission see `check_object_in_submission`

if we remove this else branch then you might as well modify `check_object_in_submission` and remove the first element from the return tuple of that function, thus eliminating the need for the `exists` check

Reply:
I made the change here as it wasn't too big of a refactor. See if it makes sense still.